autofaiss/readers/embeddings_iterators.py

```diff
@@ -170,6 +179,24 @@ def get_matrix_reader(file_format: str, fs: fsspec.AbstractFileSystem, file_path
     return cls(fs, file_path, *args)
 
 
+def filter_empty_files(file_system: fsspec.AbstractFileSystem, files: List[str], file_format: str) -> List[str]:
+    """Filter out all the empty files"""
+    if len(files) == 0:
+        return []
+    valid_files = []
+
+    def _is_empty_func(file_path):
+        return get_matrix_reader(file_format, file_system, file_path, None).is_empty(), file_path
+
+    with ThreadPool(min(16, len(files))) as pool:
+        for is_empty, file in pool.imap_unordered(_is_empty_func, files):
```

Comment:
imap_unordered is unordered, you're losing the order there

Reply:
Yes, I know. That's why I order them after the filter.

Reply:
Do you mean once in `read_total_nb_vectors_and_dim` and once in `_get_chunk_sizes` of distirbuted ?

Reply:
yes

Reply:
I don't mean that the code is duplicated, that's ok
I mean this is actually run twice and on hdfs it's not that fast to read thousand of files

Reply:
do we still need this function ?

Reply:
I removed it in the latest commit with some refactorings.